### PR TITLE
Fix the token regex to support the new format

### PR DESCRIPTION
**Pull request by droidpl**
The new format supported in tokens would make the audit log fail. https://github.blog/changelog/2021-03-04-authentication-token-format-updates/

Adding the change to support `_` on it

### CODE REVIEW

File: ghec-audit-log-utils.js [hunk 1]
```diff
@@ -15,7 +15,7 @@ function validateInput (program, config) {
   }
 
   // Validate correctness
-  const alphanumericRegex = /^[a-z0-9]+$/i
+  const tokenRegex = /^[a-z0-9_]+$/i
```
Comment (stoe) [inline ghec-audit-log-utils.js:18]: I know `/i` includes this, but the official format is `[A-Za-z0-9_]` as highlighted in https://github.blog/changelog/2021-03-04-authentication-token-format-updates/

Should we keep it the same?

**Comment by droidpl** [comment]
Updated the suggested change 😸

File: ghec-audit-log-utils.js [hunk 2]
```diff
@@ -15,7 +15,7 @@ function validateInput (program, config) {
   }
 
   // Validate correctness
-  const alphanumericRegex = /^[a-z0-9]+$/i
+  const tokenRegex = /^[A-Za-z0-9_]+$/i
```
Comment (stoe) [inline ghec-audit-log-utils.js:18]:
```suggestion
  const tokenRegex = /^[A-Za-z0-9_]+$/
```

As we're including capital- and lowercase in the regex, we don't need the `i` any longer.

Comment (stoe) [reply ghec-audit-log-utils.js:18]:
```suggestion
  const tokenRegex = /^[g(p|o|u|s|r)1_]{0,1}[A-Za-z0-9_]+$/
```

Alternatively to incorporate

- The character set is changing from `[a-f0-9]` to `[A-Za-z0-9_]`
- The format is changing to include a prefix:
  - `gp1_` for Personal Access Tokens
  - `go1_` for OAuth Access tokens
  - `gu1_` for GitHub App user-to-server tokens
  - `gs1_` for GitHub App server-to-server tokens
  - `gr1_` for GitHub App refresh tokens

Comment (droidpl) [reply ghec-audit-log-utils.js:18]: I like that suggestion, this way we ensure the format 👍

**Review by stoe — APPROVED** [review]

File: ghec-audit-log-utils.js (hunk 2 above)
Comment (stoe) [reply ghec-audit-log-utils.js]: Note: `{0,1}` is only there for backwards compatibility.

**Pull request merged**